### ISSUE TITLE
Regressions introduced on #4973

### ISSUE THREAD

**Issue by niko1point0** [RPCS3/rpcs3#5001]
In previous versions of the emulator, these games ran slow, but they ran; some of them had no graphical glitches Now in 0.0.5-7216, these game do not start at all:

Kung Fu Panda Showdown of Legendary Legends [BLES02144]
Call of Duty Modern Warfare 2 - [BLUS30377]
Call of Duty Modern Warfare 3 [BLUS30838]
Minecraft PlayStation3 Edition [BLES01976]
Blur BLES00759

**Comment by AniLeo** [comment]
Obviously, this issue needs more information

**Comment by money123451** [comment]
Can you provide a log for each of the games so kd-11 can look into also #4973 is the pr that merged in that build please use PR numbers and not appveyor build numbers.

**Comment by ihavenick** [comment]
Prince of Persia is one of them

`E {PPU[0x1000000] Thread (main_thread) [0x000b50f8]} 'sys_fs_open' failed with 0x80010012 : CELL_EISDIR, “/dev_hdd0/game/NPUB30424/USRDIR/” [1]
F {PPU[0x1000000] Thread (main_thread) [0x016351a4]} MEM: Access violation reading location 0x288`

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/2299920/RPCS3.log.gz)

**Comment by stride21** [comment]
Uncharted 2 has lower fps and  seems more unstable while in the intro or loading into game. I also had several instances where it would instantly freeze once ingame.
PR:
![216](https://user-images.githubusercontent.com/37915864/44311625-57f44d00-a3b0-11e8-89cc-0a96cc5732cc.JPG)

Previous Master:
![95](https://user-images.githubusercontent.com/37915864/44311619-414df600-a3b0-11e8-941c-a6848b81182f.JPG)

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/2300524/RPCS3.log.gz)

**Comment by Xcedf** [comment]
There are regressions for me after #4973 some games now have Fifo Queue errors and freezes like GoW 3 and games that previously were fixed of these errors like Dante's Inferno and RDR now have them again, and it seems i have a few other regressions, but i didn't investigated them yet. Will provide more information tommorow

**Comment by kd-11** [comment]
You can likely fix these by adding an empty loop to make rsx slow again. The fact that it works fine for some people and not others makes it very likely to be timing related. RDR for example works fine for me on ryzen, was able to keep going for over half an hour with no issues.

**Comment by Xcedf** [comment]
What is adding empty loop, just want to try?

**Comment by Xcedf** [comment]
Found the regression in Assassin's Creed, now game has error just before the main menu
F {rsx::thread} class std::runtime_error thrown: Assertion Failed! Vulkan API call failed with unrecoverable error: Device lost (Driver crashed with unspecified error or stopped responding and recovered) (VK_ERROR_DEVICE_LOST)
(in file c:\sources\neko\rpcs3\emu\rsx\vk\VKHelpers.h:1133)
the error came from f8a9b1fa, same commit that has graphical issues, but unlike them this one still exists in master

**Comment by TigiHof** [comment]
Silent Hill Downpour (BLUS30565) has the same error.

**Comment by kd-11** [comment]
@Xcedf If you have vulkan SDK installed, run with debug output and strict mode enabled. Its a good thing that it happens before reaching menu so log won't be too large hopefully. Attach the log you get from this.

**Comment by Xcedf** [comment]
@kd-11, with SDK installed, some kind of new errors happened in console before driver fail, anyway here's the log
[RPCS3.rar.gz](https://github.com/RPCS3/rpcs3/files/2314577/RPCS3.rar.gz)

**Comment by Xcedf** [comment]
found another regression in Killzone 2 Demo almost in the beginning at ingame, and same in Army of Two
F {rsx::thread} class std::runtime_error thrown: Verification failed: 
(in file c:\sources\neko\rpcs3\emu\rsx\rsx_cache.h:105)
can't find the exact commit, it's like issue hide itself, but can tlll that the build befor the pr is stable and in about 3-4 first commits it happens

**Comment by ruipin** [comment]
What build were you using when that last verification failure occurred? Still 7216?

**Comment by Xcedf** [comment]
Did more tests on Fifo Queue errors and freezes ln GoW 3, and they came from first commit 3b47e433
readding rsx->sync(); to rsx_methods.cpp in line 174 kinda fixes it, i ran through Kronos boss battle a several times and without a single error. And of course it cuts down the performance. Some places don't suffer from any slowdown other may have upto 20-25% fps cut.
kd-11 please take a look maybe there's a nicer way.

**Comment by kd-11** [comment]
@Xcedf You don't really need to call rsx->sync(), it justs adds an artificial stall and hides the issue. I investigated the FIFO error bug months ago, there is a latency between RSX and Cell memory transfers (it takes some time for something written by Cell to be noticed by RSX and vice versa). Modern PCs are just too fast for this. This duration is far too small to be accurately represented between PCs so I opted for a cheaper solution, just add a busy_wait counter. Unfortunately, the higher your cpu clock speed, the smaller the interval, with the payoff being that on a fast CPU other emulation components will also catch up faster. You can see the notes I made on this [here](https://github.com/RPCS3/rpcs3/blob/master/rpcs3/Emu/RSX/RSXThread.cpp#L776-L782). Try large values like 30000 and work down from there. It does hurt performance the larger the spin count (e.g 0 is very fast but more games crash even on my system)

**Comment by Xcedf** [comment]
retested Vulkan driver fail in Assassin's Creed after f8a9b1f, found out, that readding 
if (internal_width > surface_width || internal_height > surface_height) in texture_cache refixes it

**Comment by kd-11** [comment]
You'll have to grab a vulkan debug log for the assertion as that hack just evades doing the merge. Just recreate it with "Debug Output" enabled and the vulkan SDK installed.

**Comment by Xcedf** [comment]
I've already posted log to this case here in this issue, almost a month ago

**Comment by kd-11** [comment]
Right, I remember looking at this issue now. Looks like a bug in the surface_store partitioning code.

**Comment by kd-11** [comment]
OpenGL flickering is fixed by https://github.com/RPCS3/rpcs3/pull/5146

**Comment by Asinin3** [comment]
Ratchet & Clank: A Crack in Time regressed from AAA fixes. It now crashes RPCS3 with a fatal error shortly after launching the game with Write Color Buffers enabled. This is 100% reproduceable. 
![image](https://user-images.githubusercontent.com/15137035/45926653-5967e680-bf15-11e8-99fa-5d87377f6668.png)

**Comment by Xcedf** [comment]
Now that you've mentioned it i have the same error in R&C Q-Force with WCB enable, but because the game works even with WCB Disabled i did't investigted it yet

**Comment by Xcedf** [comment]
Found another issue, Resistance 3 Demo stops processing a few moments later after gatting ingame with very same reason as Assassin's Creed.

**Comment by kd-11** [comment]
So OGL works ok? Or at least doesn't crash?

**Comment by Xcedf** [comment]
Yes, OGL seems to work, no crashes

**Comment by ruipin** [comment]
Test if this works in the latest master. A fix went into my PR yesterday for segfaults caused by WCB.

**Comment by Asinin3** [comment]
It's fixed in TLoU and ACiT as well by the way as discussed on discord.

**Comment by ihavenick** [comment]
7195
![7195](https://user-images.githubusercontent.com/4084471/46093347-cdc5b280-c1bf-11e8-9665-f77c99d05912.png)
Log file
[7195.log.gz](https://github.com/RPCS3/rpcs3/files/2420585/7195.log.gz)

7216 
![7216](https://user-images.githubusercontent.com/4084471/46093530-3b71de80-c1c0-11e8-88b6-b24b4c48787b.png)
Log File
[7216.gz](https://github.com/RPCS3/rpcs3/files/2420590/7216.gz)

**Comment by kd-11** [comment]
Any regression here still left? If so, request to reopen.
